lib/foreman_maintain/concerns/downstream.rb

```diff
@@ -98,16 +98,21 @@ def product_specific_repos(rh_version_major, full_version)
       end
 
       def common_repos(rh_version_major, full_version)
-        tools_repo_id = "rhel-#{rh_version_major}-server-satellite-tools-#{full_version}-rpms"
-        maintenance_repo_id = "rhel-#{rh_version_major}-server-satellite-maintenance-6-rpms"
+        repos_arrary = common_repos_array(rh_version_major, full_version)
+        return repos_arrary.first(1) if feature(:capsule)
 
-        # Override to use Beta repositories for sat version until GA
-        if ENV['FOREMAN_MAINTAIN_USE_BETA'] == '1'
-          tools_repo_id = "rhel-#{rh_version_major}-server-satellite-tools-6-beta-rpms"
-          maintenance_repo_id = "rhel-#{rh_version_major}-server-satellite-maintenance-6-beta-rpms"
-        end
+        repos_arrary
+      end
+
+      def common_repos_array(rh_version_major, full_version)
```

Comment:
What @kgaikwad suggests looks more readable. I'd maybe go a bit further and get rid of the extra return in the middle. I'd keep use_beta as it may get handy for easier mocking.

```ruby
def common_repos(rh_version_major, full_version)
  beta_substitute = use_beta_repos? ? '-beta' : ''
  repos_list = ["rhel-#{rh_version_major}-server-satellite-maintenance-6#{beta_substitute}-rpms"]
  repos_list << "rhel-#{rh_version_major}-server-satellite-tools-#{full_version}#{beta_substitute}-rpms" unless feature(:capsule)
  repos_list
end

def use_beta_repos?
  ENV['FOREMAN_MAINTAIN_USE_BETA'] == '1'
end
```